Backpack/src/main/java/net/skyscanner/backpack/calendar2/data/CalendarCell.kt

```diff
@@ -21,28 +21,44 @@ package net.skyscanner.backpack.calendar2.data
 import java.util.Locale
 import net.skyscanner.backpack.calendar2.CalendarParams
 import net.skyscanner.backpack.calendar2.CalendarSelection
+import net.skyscanner.backpack.calendar2.extension.yearMonthHash
 import org.threeten.bp.LocalDate
 import org.threeten.bp.YearMonth
 import org.threeten.bp.format.TextStyle
+import org.threeten.bp.temporal.ChronoField
 
 internal sealed class CalendarCell {
 
   abstract val yearMonth: YearMonth
+
+  abstract val id: Long
 }
 
 internal data class CalendarSpace(
   val selected: Boolean,
   override val yearMonth: YearMonth,
-) : CalendarCell()
+) : CalendarCell() {
+
+  override val id: Long =
```

Comment:
if we don't need a "global" id for the adapter we could also first check the class before comparing the id with diffutil and then we won't have to the math here

Reply:
We're also using the ID to determine whether the items the same in DiffUtil

Reply:
that's what i meant - in `DiffUtil` we can first check if it's the same type, then compare the id. This would be an advantage of removing the stable id implementation as this way of generating the ids seems error prone if we extend it in the future

Reply:
I'm actually confused how this works with the spacers as there's multiple spacers who have the same id as they have the same `YearMonth`, right?